File: #1-25 (C, Java, Python)/Java/Euler19.java
//Euler19 ~ Counting Sundays
import java.util.concurrent.TimeUnit;

public class Euler19
{
    public static void main(String[] args)
	{
		long startTime = System.nanoTime();
		
		int day = 3; //Jan 1st, 1901 is a Tuesday
        int sundays = 0;
		int month = 1;
        int monthDays = 1;
        int year = 1901;
        while(year <= 2000)
		{
            switch(month) //Case for each month
			{
                case 1:                   //January
                    if(monthDays == 31) 
					{
                        month++;
                        monthDays = 0;
                    }
                    break;
                case 2:                                   //February
                    if((year % 4 == 0) && (year != 1900)) //Leap days
					{
                        if(monthDays == 29)
						{
                            month++;
                            monthDays = 0;
                        }
                    }
					else
					{
                        if(monthDays == 28)
						{
                            month++;
                            monthDays = 0;
                        }
                    }
                    break;
                case 3:                 //March
                    if(monthDays == 31)
					{
                        month++;
                        monthDays = 0;
                    }
                    break;
                case 4:                //April
                    if(monthDays == 30)
					{
                        month++;
                        monthDays = 0;
                    }
                    break;
                case 5:                 //May
                    if(monthDays == 31)
					{
                        month++;
                        monthDays = 0;
                    }
                    break;
                case 6:                 //June
                    if(monthDays == 30)
					{
                        month++;
                        monthDays = 0;
                    }
                    break;
                case 7:                 //July
                    if(monthDays == 31)
					{
                        month++;
                        monthDays = 0;
                    }
                    break;
                case 8:                 //August
                    if(monthDays == 31)
					{
                        month++;
                        monthDays = 0;
                    }
                    break;
                case 9:                 //September
                    if(monthDays == 30)
					{
                        month++;
                        monthDays = 0;
                    }
                    break;
                case 10:                //October
                    if(monthDays == 31)
					{
                        month++;
                        monthDays = 0;
                    }
                    break;
                case 11:                //November
                    if(monthDays == 30)
					{
                        month++;
                        monthDays = 0;
                    }
                    break;
                case 12:                //December
                    if(monthDays == 31)
					{
                        month = 1;
						monthDays = 0;
                        year++;
                    }
                    break;
            }
            
            if(day == 1 && monthDays == 1) //Sunday is 1st day of the week
			{
                sundays++;
            }
			
            if(day == 7)
			{
                day = 0;
            }
            day++;
            monthDays++;
        }
		
		long endTime = System.nanoTime();
		long timeElapsed = endTime - startTime;
		System.out.println("Solution: " + sundays + "\n");
		System.out.println("Program completed in " + timeElapsed/1000000000 + " seconds.");
    }
}

File: #1-25 (C, Java, Python)/Python/Euler19.py
#Euler19 ~ Counting Sundays
import timeit

def switch(month):
    switcher = {
        1: January,
        2: February,
        3: March,
        4: April,
        5: May,
        6: June,
        7: July,
        8: August,
        9: September,
        10: October,
        11: November,
        12: December
    }
    func = switcher.get(month)
    func()

def January():
    global month, monthDays
    if (monthDays == 31):
        month += 1
        monthDays = 0
    
def February():
    global month, monthDays, year
    if (((year % 4) == 0) & ((year != 1900))): 
        if (monthDays == 29): #Leap days
            month += 1
            monthDays = 0
    else:
        if (monthDays == 28):
            month += 1
            monthDays = 0
    
def March():
    global month, monthDays
    if (monthDays == 31):
        month += 1
        monthDays = 0

def April():
    global month, monthDays
    if (monthDays == 30):
        month += 1
        monthDays = 0

def May():
    global month, monthDays
    if (monthDays == 31):
        month += 1
        monthDays = 0

def June():
    global month, monthDays
    if (monthDays == 30):
        month += 1
        monthDays = 0

def July():
    global month, monthDays
    if (monthDays == 31):
        month += 1
        monthDays = 0

def August():
    global month, monthDays
    if (monthDays == 31):
        month += 1
        monthDays = 0

def September():
    global month, monthDays
    if (monthDays == 30):
        month += 1
        monthDays = 0

def October():
    global month, monthDays
    if (monthDays == 31):
        month += 1
        monthDays = 0

def November():
    global month, monthDays
    if (monthDays == 30):
        month += 1
        monthDays = 0

def December():
    global month, monthDays, year
    if (monthDays == 31):
        month = 1
        monthDays = 0
        year += 1

start = timeit.default_timer()

day = 3 #Jan 1st, 1901 is a Tuesday
sundays = 0
month = 1
monthDays = 1
year = 1901

while (year <= 2000):
    switch(month)
    if ((day == 1) & (monthDays == 1)): #Sunday is 1st day of the week
        sundays += 1
    
    if (day == 7):
        day = 0
    
    day += 1
    monthDays += 1

stop = timeit.default_timer()
print("Solution: ",sundays,"\n")
print("Program completed in","{0:.3f}".format(stop - start), "seconds.")

    
    
    
    
    
    
    
    
    
    
    
    
    
    
    
    
    
    
    